sdk/eventhub/azure-eventhub/azure/eventhub/_eventprocessor/in_memory_checkpoint_store.py

```diff
@@ -0,0 +1,135 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# -----------------------------------------------------------------------------------
+from typing import List, Dict, Any, Iterable, Optional, Union
+import time
+import uuid
+import logging
+
+from .checkpoint_store import CheckpointStore
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class _TrieNode(object):
+    def __init__(self, value, is_leaf):
+        self.value = value
+        self.is_leaf = is_leaf
+        if is_leaf:
+            self.children = None
+        else:
+            self.children = {}
+
+    def __contains__(self, item):
+        return item in self.children
```

Comment:
return (item in self.children) if self.children else False?

Reply:
Changed to raise an error if calling it with a leaf node